### PR TITLE
Create api endpoints for embeddings

### PR DESCRIPTION
**Pull request by Tomislaw**
Linked issue: https://github.com/Open-Model-Initiative/OMI-Data-Pipeline/issues/8

### CODE REVIEW

**Comment by Tomislaw** [comment]
Searching using image vector embedding is missing.

I have to wait until release of fastembed library with this commit is done - https://github.com/qdrant/fastembed/commit/9c72d2f59f91f87753da07c12a6f1082e233ecb3

I would be glad if you have any suggestions to current code.

File: modules/odr_core/odr_core/config.py
Comment (fearnworks) [inline modules/odr_core/odr_core/config.py:39]: Can we add an environment var for this instead of going from the project root?

File: modules/odr_core/odr_core/crud/embedding.py
Comment (fearnworks) [inline modules/odr_core/odr_core/crud/embedding.py:204]: What is the thought behind the optional here?

File: modules/odr_core/odr_core/crud/embedding.py
Comment (fearnworks) [inline modules/odr_core/odr_core/crud/embedding.py:233]: Assuming this optional is related to not being able to find the embedding. Should likely return some sort of error.

**Review by fearnworks — COMMENTED** [review]
What would it take to get some sort of smoke test that performs some simple crud cycle of an example embedding into the database?

**Comment by fearnworks** [comment]
> Searching using image vector embedding is missing.
> 
> I have to wait until release of fastembed library with this commit is done - [qdrant/fastembed@9c72d2f](https://github.com/qdrant/fastembed/commit/9c72d2f59f91f87753da07c12a6f1082e233ecb3)
> 
> I would be glad if you have any suggestions to current code.

Sounds like you think the turn around for the new release should be short?

**Comment by deskup1** [comment]
> > Searching using image vector embedding is missing.
> > I have to wait until release of fastembed library with this commit is done - [qdrant/fastembed@9c72d2f](https://github.com/qdrant/fastembed/commit/9c72d2f59f91f87753da07c12a6f1082e233ecb3)
> > I would be glad if you have any suggestions to current code.
> 
> Sounds like you think the turn around for the new release should be short?

It looks like they have frequentreleases, somewhere about once per month.

If we don't want to wait, I can try to specify requirements to use version from master branch of their repository.

**Comment by takoyaro** [comment]
Overall, this looks good, and I share the same questions as fearnworks. 
I suggest we merge this once image embedding is functional. If that requires using a specific branch of fastembed, let's proceed with that for now. We can switch back to main once qdrant merges the PR.

File: modules/odr_core/odr_core/crud/embedding.py
Comment (Tomislaw) [reply modules/odr_core/odr_core/crud/embedding.py]: I used the same convention as for other get crud operations.

File: modules/odr_core/odr_core/crud/embedding.py
Comment (Tomislaw) [reply modules/odr_core/odr_core/crud/embedding.py]: Yeah, It doesn't make sense.
Fixed it.

File: modules/odr_core/odr_core/config.py
Comment (Tomislaw) [reply modules/odr_core/odr_core/config.py:39]: Added.

**Comment by Tomislaw** [comment]
> What would it take to get some sort of smoke test that performs some simple crud cycle of an example embedding into the database?

I believe creating smoke tests should be done as part of another PR

I can help anyone who will be responsible for creating smoke tests in creating tests for annotation endpoints

**Review by hipsterusername — APPROVED** [review]

**Pull request merged**
